src/test/java/com/sap/cloudfoundry/client/facade/CloudStackIntegrationTest.java

```diff
@@ -0,0 +1,122 @@
+import com.sap.cloudfoundry.client.facade.domain.CloudRouteSummary;
+import com.sap.cloudfoundry.client.facade.domain.ImmutableCloudApplication;
+import com.sap.cloudfoundry.client.facade.domain.ImmutableCloudRouteSummary;
+import com.sap.cloudfoundry.client.facade.domain.ImmutableStaging;
+import com.sap.cloudfoundry.client.facade.domain.Staging;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Set;
+
+import static com.sap.cloudfoundry.client.facade.IntegrationTestConstants.APPLICATION_HOST;
+import static com.sap.cloudfoundry.client.facade.IntegrationTestConstants.DEFAULT_DOMAIN;
+import static com.sap.cloudfoundry.client.facade.IntegrationTestConstants.DISK_IN_MB;
+import static com.sap.cloudfoundry.client.facade.IntegrationTestConstants.MEMORY_IN_MB;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+class CloudStackIntegrationTest extends CloudControllerClientIntegrationTest {
+
+    private static final String DEFAULT_STACK_NAME = "cflinuxfs3";
+    private static final String INCORRECT_STACK_NAME = "Non-existent-stack-name";
+    private static final CloudRouteSummary DEFAULT_ROUTE_SUMMARY = ImmutableCloudRouteSummary.builder()
+                                                                                             .host(APPLICATION_HOST)
+                                                                                             .domain(DEFAULT_DOMAIN)
+                                                                                             .build();
+
+    @BeforeAll
+    static void createDefaultDomain() {
+        client.addDomain(DEFAULT_DOMAIN);
+    }
+
+    @AfterAll
+    static void deleteDefaultDomain() {
+        List<CloudRoute> routes = client.getRoutes(DEFAULT_DOMAIN);
+        for (CloudRoute route : routes) {
+            client.deleteRoute(route.getHost(), DEFAULT_DOMAIN, null);
+        }
+        client.deleteDomain(DEFAULT_DOMAIN);
+    }
+
+    @Test
+    @DisplayName("Create application and verify default cloud stack")
+    void createApplicationWithDefaultCloudStack() {
+        String applicationName = "test-stack-app-1";
+        Staging staging = ImmutableStaging.builder()
+                                          .addBuildpack(IntegrationTestConstants.JAVA_BUILDPACK)
+                                          .build();
+        try {
+            verifyExistenceOfDefaultStack(applicationName, staging, Set.of(DEFAULT_ROUTE_SUMMARY));
+        } finally {
+            client.deleteApplication(applicationName);
+        }
+    }
+
+    private void verifyExistenceOfDefaultStack(String applicationName, Staging staging, Set<CloudRouteSummary> cloudRoutesSummary) {
```

Comment:
These are multiple cloud routes, not a single summary of many routes. Rename the 3rd variable